File: mobile/lib/features/architect/domains/use_cases/team/update.dart
import 'package:dartz/dartz.dart';
import 'package:equatable/equatable.dart';

import '../../../../../core/errors/failure.dart';
import '../../../../../core/use_cases/usecase.dart';
import '../../entities/team.dart';
import '../../repositories/team.dart';

class TeamUpdate implements UseCase<Team, Params> {
  final TeamRepository repository;

  TeamUpdate(this.repository);

  @override
  Future<Either<Failure, Team>> call(Params params) async {
    return await repository.update(
      title: params.title,
      description: params.description,
      image: params.image,
      teamId: params.teamId,
      members: params.members,
    );
  }
}

class Params extends Equatable {
  final String title;
  final String? description;
  final String? image;
  final String teamId;
  final List<String>? members;

  const Params({
    required this.title,
    required this.teamId,
    this.description,
    this.image,
    this.members,
  });

  @override
  List<Object?> get props => [title, description, image, teamId];
}

File: backend/api/app/domain/use_cases/team/update.py
from core.use_cases.use_case import UseCase
from app.domain.repositories.team import BaseRepository
from core.common.equatable import Equatable
from core.common.either import Either
from core.errors.failure import Failure
from app.domain.entities.team import TeamEntity, Team

class Params(Equatable):
    def __init__(self, team: Team, team_id: str, user_id: str) -> None:
        self.team = team
        self.team_id = team_id
        self.user_id = user_id

class UpdateTeam(UseCase[TeamEntity]):
    def __init__(self, repository: BaseRepository):
        self.repository = repository
    
    async def __call__(self, params: Params) -> Either[Failure, TeamEntity]:
        return await self.repository.update_team(params.team, params.team_id, params.user_id)

